src/Aspire.Hosting.Docker/DockerComposePublishingContext.cs

```diff
@@ -308,17 +314,17 @@ private void ProcessVolumes()
             }
         }
 
-        private string GetValue(EndpointMapping mapping, EndpointProperty property)
+        private static string GetValue(EndpointMapping mapping, EndpointProperty property)
         {
-            var (scheme, host, port, targetPort, isHttpIngress, _) = mapping;
+            var (scheme, host, internalPort, exposedPort, isHttpIngress, _) = mapping;
 
             return property switch
             {
-                EndpointProperty.Url => GetHostValue($"{scheme}://", suffix: isHttpIngress ? null : $":{port}"),
+                EndpointProperty.Url => GetHostValue($"{scheme}://", suffix: isHttpIngress ? null : $":{internalPort}"),
                 EndpointProperty.Host or EndpointProperty.IPV4Host => GetHostValue(),
-                EndpointProperty.Port => port.ToString(CultureInfo.InvariantCulture),
-                EndpointProperty.HostAndPort => GetHostValue(suffix: $":{port}"),
-                EndpointProperty.TargetPort => $"{targetPort ?? composePublishingContext.PortAllocator.AllocatePort()}",
+                EndpointProperty.Port => internalPort.ToString(CultureInfo.InvariantCulture),
```

Comment:
This is the exposed port, not internal port.

Reply:
Ohhh all container to container comms

Reply:
Yep container to container, doesn't use the host port needs the internal as its internal to the docker network
It was wrong before ^^